sdk/internal/errorinfo/errorinfo_test.go

```diff
@@ -0,0 +1,39 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package errorinfo
+
+import (
+	"testing"
+)
+
+func TestNonRetriableError(t *testing.T) {
+	//Create sample error.
+	err := NonRetriableError(NewSampleError("Do Not Retry"))
+
+	// Check if the error is of type NonRetriableError
+	if _, ok := err.(*nonRetriableError); !ok {
```

Comment:
The concrete type isn't necessarily the important part - it's just an implementation detail. You'll want to check that your type conforms to the "NonRetriable" shape by using an interface.

For example:

```go
var nonRetriableErr *interface { NonRetriable() }
require.ErrorAs(err, &nonRetriableErr)
```

Reply:
Or use the predefined interface type.
```go
var nonRetriableErr *NonRetriable
```